docs/DesignDocs/AttestationApiRequirements.md

```diff
@@ -0,0 +1,291 @@
+Proposed Attestation API Requirements
+=====================================
+
+This design document proposes requirements for APIs that applications can
+use for performing attestation of enclave applications.
+
+Motivation
+----------
+
+Requirements:
+* Usable by Attesters, Verifiers, and Relying Parties
+* Able to support different TEEs
+* Able to support different protocols
+* Able to support different topological models
+  (e.g., Passport model vs Background-check model)
+* Able to support different Evidence formats
+* Able to support different Attestation Results formats
+* Able to support different Verifier services, including a local Verifier
```

Comment:
Is this composite device, with multiple roles in the same device / security domain?

Reply:
Not sure what you mean by "this".  Today for example, oe_verify_report was written to _only_ support a combined Verifier+Relying Party.  These abstract APIs are more general, with no such restriction.